.changeset/bright-seas-march.md

```diff
@@ -0,0 +1,7 @@
+---
+'@leafygreen-ui/confirmation-modal': minor
+---
+
+- Dropped the `isRequired` from the `buttonText` prop type. As of [version 5.1.0](https://github.com/mongodb/leafygreen-ui/blob/main/packages/confirmation-modal/CHANGELOG.md#510), `buttonText` was marked as a deprecated prop that is now optional.
+- Fixed existing confirm button disabled state logic. When modal is closed and reopened when interacting with cancel or confirm button, confirm button will conditionally reset to disabled state only when `requiredInputText` is provided.
+- Added missing confirm button disabled state logic. gWhen modal is closed and reopened when interacting with modal close button, confirm button will conditionally reset to disabled state when `requiredInputText` is provided.
```

Comment:
Is this repeating the bullet above?

Reply:
updated the language for clarity. 2nd bullet addresses fixing cancel/confirm button while the 3rd bullet addresses the missing logic for the modal close button ([this line](https://github.com/mongodb/leafygreen-ui/pull/2367/files#diff-28b99ffb5f3e5a00a3ae8ac9b42c39632b5973489941fc206fade38025a31dabL97-R102))